ogre2/src/Ogre2GpuRays.cc

```diff
@@ -697,7 +723,42 @@ void Ogre2GpuRays::Setup1stPass()
     colorTexDef->preferDepthTexture = true;
     colorTexDef->fsaaExplicitResolve = false;
 
-    nodeDef->setNumTargetPass(2);
+    Ogre::TextureDefinitionBase::TextureDefinition *particleDepthTexDef =
+        nodeDef->addTextureDefinition("particleDepthTexture");
+    particleDepthTexDef->textureType = Ogre::TEX_TYPE_2D;
+    particleDepthTexDef->width = 0;
+    particleDepthTexDef->height = 0;
+    particleDepthTexDef->depth = 1;
+    particleDepthTexDef->numMipmaps = 0;
+    particleDepthTexDef->widthFactor = 0.5;
+    particleDepthTexDef->heightFactor = 0.5;
+    particleDepthTexDef->formatList = {Ogre::PF_D32_FLOAT};
+    particleDepthTexDef->fsaa = 0;
+    particleDepthTexDef->uav = false;
+    particleDepthTexDef->automipmaps = false;
+    particleDepthTexDef->hwGammaWrite = Ogre::TextureDefinitionBase::BoolFalse;
+    particleDepthTexDef->depthBufferId = Ogre::DepthBuffer::POOL_DEFAULT;
+
+    Ogre::TextureDefinitionBase::TextureDefinition *particleTexDef =
+        nodeDef->addTextureDefinition("particleTexture");
+    particleTexDef->textureType = Ogre::TEX_TYPE_2D;
+    particleTexDef->width = 0;
+    particleTexDef->height = 0;
+    particleTexDef->depth = 1;
+    particleTexDef->numMipmaps = 0;
+    particleTexDef->widthFactor = 0.5;
+    particleTexDef->heightFactor = 0.5;
+    particleTexDef->formatList = {Ogre::PF_R8G8B8};
+    particleTexDef->fsaa = 0;
+    particleTexDef->uav = false;
+    particleTexDef->automipmaps = false;
+    particleTexDef->hwGammaWrite = Ogre::TextureDefinitionBase::BoolFalse;
+    particleTexDef->depthBufferId = Ogre::DepthBuffer::POOL_DEFAULT;
+    particleTexDef->depthBufferFormat = Ogre::PF_D32_FLOAT;
+    particleTexDef->preferDepthTexture = true;
+    particleTexDef->fsaaExplicitResolve = false;
+
+    nodeDef->setNumTargetPass(3);
```

Comment:
How did we go from 2 passes to 3? What are the 3 passes here?

Reply:
went from 

```
target colorTexture
target rt_input
```
to

```
target colorTexture
target particleTexture
target rt_input
```

This was easier to do compared to depth camera as I didn't have to worry about color texture.

Reply:
I see that now. I was confusing the target passes (3) with the texture definitions (4). Thanks :+1: